### PR TITLE
statistics: use TopN

### PR DESCRIPTION
**Pull request by Luffbee**
### What problem does this PR solve? 

Close issue #1997 .

### What is changed and how it works?

Replace `TwoQueueCache` with `TopN`.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/statistics/hot_peer.go [hunk 1]
```diff
@@ -41,6 +41,17 @@ type HotPeerStat struct {
 	isNew      bool
 }
 
+// ID returns region ID. Implementing TopNItem.
+func (stat *HotPeerStat) ID() uint64 {
+	return stat.RegionID
+}
+
+// Less compares two HotPeerStat.Implementing TopNItem.
+func (stat *HotPeerStat) Less(than TopNItem) bool {
```
Comment (lhy1024) [inline server/statistics/hot_peer.go]: `than` look like strange

Comment (Luffbee) [reply server/statistics/hot_peer.go]: Learned it from google btree https://godoc.org/github.com/google/btree#Item.

**Pull request merged**